### PR TITLE
LibGUI: Implement vim motions for LeftBrace and RightBrace

### PR DESCRIPTION
**Pull request by scwfri**
In VimEditingEngine, implemented vim motions for Key_LeftBrace and Key_RightBrace

### CODE REVIEW

File: Userland/Libraries/LibGUI/VimEditingEngine.cpp [hunk 1]
```diff
@@ -971,13 +971,22 @@ bool VimEditingEngine::on_key_in_normal_mode(const KeyEvent& event)
                 move_one_up(event);
                 switch_to_insert_mode();
                 return true;
-            // FIXME: Integrate these into vim motions too.
-            case (KeyCode::Key_LeftBrace):
-                move_to_previous_empty_lines_block();
+            case (KeyCode::Key_LeftBrace): {
+                auto amount = m_motion.amount() ? m_motion.amount() : 1;
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibGUI/VimEditingEngine.cpp:975]: Consider making this behavior more obvious:
```suggestion
                auto amount = m_motion.amount() > 0 ? m_motion.amount() : 1;
```

File: Userland/Libraries/LibGUI/VimEditingEngine.cpp [hunk 2]
```diff
@@ -971,13 +971,22 @@ bool VimEditingEngine::on_key_in_normal_mode(const KeyEvent& event)
...
+                m_motion.reset();
+                for (int i = 0; i < amount; i++) {
+                    move_to_previous_empty_lines_block();
+                }
                 return true;
-            case (KeyCode::Key_RightBrace):
-                move_to_next_empty_lines_block();
+            }
+            case (KeyCode::Key_RightBrace): {
+                auto amount = m_motion.amount() ? m_motion.amount() : 1;
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibGUI/VimEditingEngine.cpp:983]: (see above)

File: Userland/Libraries/LibGUI/VimEditingEngine.cpp (hunk 1 above)
Comment (scwfri) [reply Userland/Libraries/LibGUI/VimEditingEngine.cpp]: Agreed that makes sense; updated

**Review by kleinesfilmroellchen — APPROVED** [review]

File: Userland/Libraries/LibGUI/VimEditingEngine.cpp [hunk 3]
```diff
@@ -971,13 +971,22 @@ bool VimEditingEngine::on_key_in_normal_mode(const KeyEvent& event)
...
                 return true;
-            // FIXME: Integrate these into vim motions too.
-            case (KeyCode::Key_LeftBrace):
-                move_to_previous_empty_lines_block();
+            case (KeyCode::Key_LeftBrace): {
+                auto amount = m_motion.amount() > 0 ? m_motion.amount() : 1;
+                m_motion.reset();
+                for (int i = 0; i < amount; i++) {
+                    move_to_previous_empty_lines_block();
+                }
```
Comment (linusg) [inline Userland/Libraries/LibGUI/VimEditingEngine.cpp:979]: please omit curly braces for single-line if/for/while etc. bodies

Comment (scwfri) [reply Userland/Libraries/LibGUI/VimEditingEngine.cpp]: Updated.

**Pull request merged**